src/AWS.Deploy.Orchestration/RecipeHandler.cs

```diff
@@ -12,24 +15,37 @@ namespace AWS.Deploy.Orchestration
 {
     public class RecipeHandler
     {
-        public static List<RecipeDefinition> GetRecipeDefinitions()
+        public static async Task<List<RecipeDefinition>> GetRecipeDefinitions(ICustomRecipeLocator customRecipeLocator, ProjectDefinition? projectDefinition)
         {
-            var recipeDefinitionsPath = RecipeLocator.FindRecipeDefinitionsPath();
+            IEnumerable<string> recipeDefinitionsPaths = new List<string> { RecipeLocator.FindRecipeDefinitionsPath() };
+            if(projectDefinition != null)
+            {
+                var targetApplicationFullPath = new DirectoryInfo(projectDefinition.ProjectPath).FullName;
+                var solutionDirectoryPath = !string.IsNullOrEmpty(projectDefinition.ProjectSolutionPath) ?
+                    new DirectoryInfo(projectDefinition.ProjectSolutionPath).Parent.FullName : string.Empty;
+
+                var customPaths = await customRecipeLocator.LocateCustomRecipePaths(targetApplicationFullPath, solutionDirectoryPath);
+                recipeDefinitionsPaths = recipeDefinitionsPaths.Union(customPaths);
+            }
+
             var recipeDefinitions = new List<RecipeDefinition>();
 
             try
             {
-                foreach (var recipeDefinitionFile in Directory.GetFiles(recipeDefinitionsPath, "*.recipe", SearchOption.TopDirectoryOnly))
+                foreach(var recipeDefinitionsPath in recipeDefinitionsPaths)
                 {
-                    try
-                    {
-                        var content = File.ReadAllText(recipeDefinitionFile);
-                        var definition = JsonConvert.DeserializeObject<RecipeDefinition>(content);
-                        recipeDefinitions.Add(definition);
-                    }
-                    catch (Exception e)
+                    foreach (var recipeDefinitionFile in Directory.GetFiles(recipeDefinitionsPath, "*.recipe", SearchOption.TopDirectoryOnly))
                     {
-                        throw new Exception($"Failed to Deserialize Recipe Definition [{recipeDefinitionFile}]: {e.Message}", e);
+                        try
+                        {
+                            var content = File.ReadAllText(recipeDefinitionFile);
+                            var definition = JsonConvert.DeserializeObject<RecipeDefinition>(content);
+                            recipeDefinitions.Add(definition);
+                        }
+                        catch (Exception e)
+                        {
+                            throw new Exception($"Failed to Deserialize Recipe Definition [{recipeDefinitionFile}]: {e.Message}", e);
```

Comment:
This exception message is too generic, if file IO fails, it will show the same message which isn't actually deserialization.

Reply:
The message also contains the underlying exception message. For an exception that shouldn't happen that should be good enough. I don't think adding a `catch(IOException e)` and then still just rethrowing with a our own message will improve things.